code/game/gamemodes/miniantags/umbra/umbra_mob.dm

```diff
@@ -0,0 +1,275 @@
+		return
+	if(!L.stat)
+		src << "<span class='warning'>[L] is conscious again and shields your efforts!</span>"
+		L << "<span class='warning'>You're being watched.</span>"
+		harvesting = FALSE
+		return
+	var/vitae_yield = 1 //A bit of essence even if it's a weak soul
+	var/vitae_information = "<span class='umbra'>This soul is "
+	if(ishuman(L))
+		vitae_information += "human, "
+		vitae_yield += rand(10, 15)
+	if(L.mind)
+		if(!(L.mind in ticker.mode.devils))
+			vitae_information += "sentient, "
+		else
+			vitae_information += "infernal and home to multiple souls, all of which you can drain! Moving on... it's "
+			for(var/i in 1 to (L.mind.devilinfo.soulsOwned.len))
+				vitae_yield += rand(5, 10) //You can drain all the souls that a devil has stolen!
+		vitae_yield += rand(10, 15)
+	if(L.stat == UNCONSCIOUS)
+		vitae_information += "blazing with vitality, "
+		vitae_yield += rand(20, 25) //Significant bonus if the target is in critical condition instead of dead
+	else if(L.stat == DEAD)
+		vitae_information += "dim but still usable, "
+		vitae_yield += rand(1, 10)
+	vitae_information += "and ready for harvest. You'll absorb around [vitae_yield] vitae - "
+	switch(vitae_yield)
+		if(0 to 15)
+			vitae_information += "not much, but everything counts."
+		if(15 to 30)
+			vitae_information += "decent, but not great."
+		if(30 to 45)
+			vitae_information += "about what you could expect."
+		if(45 to 60)
+			vitae_information += "a good bit, more than you've come to expect."
+		if(75 to vitae_cap)
+			vitae_information += "<i>a bounty, more than you could ever dream of!</i>"
+	vitae_information += " Now, for the harvest...</span>"
+	src << vitae_information
+	if(!do_after(src, 30, target = L))
+		harvesting = FALSE
+		return
+	if(L.hellbound)
+		src << "<span class='warning'>[L] seems to have lost their soul!</span>"
+		harvesting = FALSE
+		return
+	if(!L.stat)
+		src << "<span class='warning'>[L] is conscious again and shields your efforts!</span>"
+		L << "<span class='warning'>A chill runs across your body.</span>"
+		harvesting = FALSE
+		return
+	Stun(50)
+	Reveal(50)
+	visible_message("<span class='warning'>[src] flickers into existence and reaches out towards [L]...</span>")
+	L.visible_message("<span class='warning'>...who rises into the air, shuddering as purple light streams towards out of their body!</span>")
+	animate(L, pixel_y = pixel_y + 5, time = 10)
+	Beam(L, icon_state = "drain_life", icon = 'icons/effects/effects.dmi', time = 50)
+	sleep(50)
+	adjust_vitae(vitae_yield, FALSE, "[L]. They won't yield any more for the time being")
+	recently_drained |= L
+	visible_message("<span class='warning'>[src] winks out of existence, releasing its hold on [L]...</span>")
+	L.visible_message("<span class='warning'>...who falls unceremoniously back to the ground.</span>")
+	animate(L, pixel_y = pixel_y - 5, time = 10)
+	spawn(rand(UMBRA_MAX_HARVEST_COOLDOWN - 600, UMBRA_MAX_HARVEST_COOLDOWN)) //Between 4-5 minutes by default
```

Comment:
Here, you would use the following syntax:
```
var/time_to_call = rand(UMBRA_MAX_HARVEST_COOLDOWN - 600, UMBRA_MAX_HARVEST_COOLDOWN)
addtimer(src, "remove_from_list", time_to_call, FALSE, L)
```
After which, I think you'd do something like:
```
/mob/living/simple_animal/umbra/proc/remove_from_list(mob/living/L)
    if(!L)
        return
    src << "<span class='umbra'>You think that [L]'s soul should be strong enough to harvest again.</span>"
    recently_drained -= L
```